gb-ppu/examples/tilemap_viewer.rs

```diff
@@ -56,9 +64,16 @@ pub fn main() {
             egui::menu::bar(ui, |ui| {
                 ui.set_height(render::MENU_BAR_SIZE);
                 egui::menu::menu(ui, "dump", |ui| {
-                    for (title, dump) in dumps {
+                    for (title, vram, io_reg) in dumps {
                         if ui.button(title).clicked() {
-                            ppu.overwrite_vram(dump);
+                            ppu.overwrite_vram(vram);
+                            *ppu.bg_palette_mut() = io_reg[0x47].into();
+                            ppu.control_mut()
+                                .set_win_tilemap_area((io_reg[0x40] & 0b0100_0000) != 0);
+                            ppu.control_mut()
+                                .set_bg_tilemap_area((io_reg[0x40] & 0b0000_1000) != 0);
+                            ppu.control_mut()
+                                .set_bg_win_tiledata_area((io_reg[0x40] & 0b0001_0000) != 0);
```

Comment:
Same here, can you put the code into an utility function ? (can be inside the example code)